src/feed/lookup/epoch.ts

```diff
@@ -0,0 +1,38 @@
+import Long from 'long'
+// import { maxuint64 } from './lookup'
+import { Bytes } from '@ethersphere/bee-js/dist/src/utils/bytes'
+
+const EpochLength = 8
+const MaxTime = Long.fromNumber(1).shiftLeft(56).sub(1)
+
+export declare type EpochID = Bytes<typeof EpochLength>
+
+export function getBaseTime(time: number, level: number): Long {
+  return Long.fromNumber(time).and(Long.MAX_UNSIGNED_VALUE.shiftLeft(level))
+}
+
+export class Epoch {
+  level
+  time
+
+  constructor(level: number, time: Long) {
+    this.level = level
+    this.time = time
+  }
+
+  base(): Long {
+    return getBaseTime(this.time, this.level)
+  }
+
+  id(): EpochID {
+    const base = this.base()
+    const id = Uint8Array.from(base.toBytes()) as EpochID
+    id[7] = this.level
```

Comment:
So I guess from this it would use only a bytelength for the epoch serialisation, the other bytes could be only padding for the correct hashing, am I wrong?

Reply:
Most likely this is the correct explanation, but the meaning is left to the author of the original golang code.